Unittest.py

```diff
@@ -337,6 +338,28 @@ def test_ganondorf(self):
                 self.assertNotEqual('Ganons Tower Boss Key Chest', spoiler.worlds[0].light_arrow_location.name)
 
 
+class TestEntranceRandomizer(unittest.TestCase):
+    def test_spawn_point_warp_reachability(self):
```

Comment:
I don't think this is a useful unittest--if it becomes possible to place a spawn in Colossus, then this test could still succeed on some versions because it's testing whether this specific seed can be built. You'd be better off testing whether a spawn is allowed to be somewhere that can't reach ToT... and since we know what the world otherwise looks like with all other ER settings off, that's essentially a hardcodeable list. (Not that you should make a unittest with a hardcoded list... just maybe a unittest that puts the spawn in Colossus and checks that it's rejected when building the graph?)

Reply:
It would be faster to directly test the function used to determine what's a valid spawn, rather than actually run the entrance placement.

Reply:
There are not any explicit limits I can see placed on spawn points other than target entrance pools. The only thing preventing a Colossus or Fountain spawn is the check for no-item Kokiri/Kakariko access in `validate_worlds`. This performs the same search process I used for Lake/ToT.

I changed the test to check that entrance shuffle error occurs with either a Colossus or Fountain adult spawn with ocarina as a starting item. If the Kokiri/Kakariko check is removed, seed generation works using a warp song in Colossus or Ice Cavern, and the unit test fails as expected. It doesn't test directly for Lake/ToT reachability, but I've add a comment with this as the reason for the test.

Reply:
That ought to suffice. Thanks.